webapp/channels/src/components/advanced_text_editor/use_key_handler.tsx

```diff
@@ -159,6 +159,13 @@ const useKeyHandler = (
         }
 
         if ((Keyboard.isKeyPressed(e, KeyCodes.PAGE_UP) || Keyboard.isKeyPressed(e, KeyCodes.PAGE_DOWN))) {
+            const textbox = textboxRef.current?.getInputBox();
+            const hasScrollableContent = textbox.scrollHeight > textbox.clientHeight;
```

Comment:
Your changes make the scrolling behave closer to how it would if the textbox was inside the scrollable area, but it's not completely accurate yet because pressing page up/down should still scroll the page if the textbox can't scroll any further. Could you change this to check if the textbox can scroll up/down as well?